gnocchi/tests/test_carbonara.py

```diff
@@ -158,11 +158,63 @@ def test_before_epoch(self):
                           ts.group_serie, 60)
 
     @staticmethod
-    def _resample(ts, sampling, agg, max_size=None):
+    def _resample(ts, sampling, agg, max_size=None, derived=False):
         grouped = ts.group_serie(sampling)
+        if derived:
+            grouped = grouped.derived()
         return carbonara.AggregatedTimeSerie.from_grouped_serie(
             grouped, sampling, agg, max_size=max_size)
 
+    def test_derived_last(self):
+        ts = carbonara.TimeSerie.from_tuples(
+            [(datetime.datetime(2014, 1, 1, 12, 0, 0), 50),
+             (datetime.datetime(2014, 1, 1, 12, 0, 4), 55),
+             (datetime.datetime(2014, 1, 1, 12, 1, 2), 65),
+             (datetime.datetime(2014, 1, 1, 12, 1, 14), 66),
+             (datetime.datetime(2014, 1, 1, 12, 1, 24), 70),
+             (datetime.datetime(2014, 1, 1, 12, 2, 4), 83),
+             (datetime.datetime(2014, 1, 1, 12, 2, 35), 92),
+             (datetime.datetime(2014, 1, 1, 12, 2, 42), 103),
+             (datetime.datetime(2014, 1, 1, 12, 3, 2), 105),
+             (datetime.datetime(2014, 1, 1, 12, 3, 22), 5),  # Counter reset
+             (datetime.datetime(2014, 1, 1, 12, 3, 42), 7),
+             (datetime.datetime(2014, 1, 1, 12, 4, 9), 23)])
+        ts = self._resample(ts, 60, 'mean', derived=True)
+
+        self.assertEqual(5, len(ts))
+        self.assertEqual(
+            [(datetime.datetime(2014, 1, 1, 12, 0, 0), 60, 5),
+             (datetime.datetime(2014, 1, 1, 12, 1, 0), 60, 5),
```

Comment:
OK thanks for the explanation, I forgot to compute `2014-1-1 12:01:02 65 10` in my mind :) That all makes sense. I think keeping the ending timestamp makes more sense. Let's go that way for now, we can add yet-another-agg-method if needed later.